File: typescript/raindrops/raindrops.ts
const MAP = {3: "Pling", 5: "Plang", 7: "Plong"}

export const convert = (n: number) =>
    Object.entries(MAP).filter(([k]) => n % parseInt(k) === 0).map(([_, v]) => v).join("") || n.toString();

File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    let str = [(3,"Pling");(5,"Plang");(7,"Plong")]
            |> Seq.map (fun (k, v) -> if number % k = 0 then v else "")
            |> String.concat ""
    if str <> "" then str else number.ToString()
